pkg/helm/io.go

```diff
@@ -38,10 +40,26 @@ func ReplaceValuesInHelmValuesFile(values map[string]string, yamlPath string) (s
 		return "", err
 	}
 
-	fileContent := string(readFile)
-	for k, v := range values {
-		fileContent = strings.ReplaceAll(fileContent, k, v)
+	var buffer bytes.Buffer
+
+	scanner := bufio.NewScanner(bytes.NewReader(readFile))
+	for scanner.Scan() {
+		line := scanner.Text()
+
+		for k, v := range values {
+			re := regexp.MustCompile(`\b` + regexp.QuoteMeta(k) + `$`)
```

Comment:
Comment explaining the error case

i.e.

Source:

```
ghcr.io/openfaas/faas-netes:0.1.0-rc
ghcr.io/openfaas/faas-netes:0.1.0
```

Replaces both substrings with "ghcr.io/openfaas/faas-netes:0.2.0"

Could you also add a unit test to prove it doesn't update both